In [1]:
# required modules
import pandas as pd
import numpy as np 
import matplotlib.pyplot as plt
%matplotlib inline
plt.style.use('default')
pd.set_option('display.max_columns', None)

In [2]:
df_rest = pd.read_csv('../best_restaurant_location/data/data_combined_v1.00.csv')
df_rest.head()

place_id  tripadvisor_id                    fsq_id  \
0  ChIJd6kh67ZkjEcRC1IQourQOyo       7159041.0  4de685281f6e9d90c7f5325c   
1  ChIJX68uubZkjEcRKKxpZ7OrgqU      13322506.0                       NaN   
2  ChIJPcYsG_BljEcR4kTFqBJJoK4       6720272.0                       NaN   
3  ChIJhb4e-Ep7jEcRY38kziIGZmE             NaN                       NaN   
4  ChIJV8uySFxljEcRn7XpIO3zZX0             NaN                       NaN   

                              name business_status  \
0          Restaurant La Tentation     OPERATIONAL   
1            Café Restaurant Palma     OPERATIONAL   
2                    Le Versailles     OPERATIONAL   
3                 Restaurant Divan     OPERATIONAL   
4  Restaurant Les Jardins du Rhône     OPERATIONAL   

                                               types  price_level  \
0  ['restaurant', 'point_of_interest', 'food', 'e...          2.0   
1  ['restaurant', 'point_of_interest', 'food', 'e...          NaN   
2  ['restaurant', 'point_of_interest', 'food', 'e...          NaN   
3  ['restaurant', 'food', 'point_of_interest', 'e...          2.0   
4  ['restaurant', 'food', 'point_of_interest', 'e...          NaN   

   ta_price_level  rating  user_ratings_total  ta_rating  \
0             2.0     4.2               158.0        3.5   
1             NaN     3.7               141.0        2.0   
2             2.0     3.9                56.0        4.0   
3             NaN     4.5               113.0        NaN   
4             NaN     NaN                 NaN        NaN   

   ta_user_ratings_total  combined_rating  geometry.location.lat  \
0                   34.0              4.1              46.205232   
1                    3.0              3.7              46.204931   
2                    5.0              3.9              46.205752   
3                    NaN              4.5              46.196924   
4                    NaN              NaN              46.205248   

   geometry.location.lng                                  formatted_address  \
0               6.114945            Av. d'Aïre 91, 1203 Genève, Switzerland   
1               6.114021         Av. d'Aïre 93bis, 1203 Genève, Switzerland   
2               6.114763       Chem. des Sports 2, 1203 Genève, Switzerland   
3               6.115081  Rte de Saint-Georges 81, 1213 Petit-Lancy, Swi...   
4               6.115503            Av. d'Aïre 89, 1203 Genève, Switzerland   

   zip_code         city international_phone_number  \
0      1203       Genève           +41 22 344 28 29   
1      1203       Genève           +41 22 345 84 48   
2      1203       Genève           +41 22 344 49 41   
3      1213  Petit-Lancy           +41 22 792 96 80   
4      1203       Genève           +41 22 561 38 04   

                                            website     google_category  \
0                                               NaN          Restaurant   
1                                               NaN          Restaurant   
2  https://yellow.local.ch/d/iWjaP0dHU9GYe4GDduNgnw  Italian restaurant   
3                                               NaN          Restaurant   
4                                               NaN          Restaurant   

  ta_4sq_cuisine1 ta_4sq_cuisine2 ta_4sq_cuisine3 ta_4sq_cuisine4  \
0        European           Swiss   International               0   
1               0               0               0               0   
2         Italian          French   International        European   
3               0               0               0               0   
4               0               0               0               0   

    ta_4sq_cuisine5       main_category_1 main_category_2 main_category_3  \
0                 0              European           Swiss             NaN   
1                 0  General / Restaurant             NaN             NaN   
2  Central European               Italian          French        European   
3                 0  General / Restaurant             NaN             NaN  

In [3]:
df_rest.columns

Index(['place_id', 'tripadvisor_id', 'fsq_id', 'name', 'business_status',
       'types', 'price_level', 'ta_price_level', 'rating',
       'user_ratings_total', 'ta_rating', 'ta_user_ratings_total',
       'combined_rating', 'geometry.location.lat', 'geometry.location.lng',
       'formatted_address', 'zip_code', 'city', 'international_phone_number',
       'website', 'google_category', 'ta_4sq_cuisine1', 'ta_4sq_cuisine2',
       'ta_4sq_cuisine3', 'ta_4sq_cuisine4', 'ta_4sq_cuisine5',
       'main_category_1', 'main_category_2', 'main_category_3',
       'main_category_4', 'combined_main_category', 'sub_category',
       'neighboorhood_area_km2', 'resident_population_total',
       'population_age_0_14', 'population_15_19', 'population_20_24',
       'population_25_34', 'population_35_44', 'population_45_54',
       'population_54_64', 'population_65+', 'resident_per_km2',
       'brut_income_annual_per_person', 'cluster'],
      dtype='object')

## Restaurant Type and Cuisine type

In [4]:
# only two restaurant do not have any category
df_rest[df_rest['combined_main_category'].isnull()] 

place_id  tripadvisor_id fsq_id  \
925   ChIJHVcqMPZljEcRc0nUGeodWtI             NaN    NaN   
1305  ChIJud7aEC5ljEcR69lcbLgrBBQ       9744737.0    NaN   

                       name     business_status  \
925   Cave Restaurant Le180      FUTURE_OPENING   
1305      Bistrot Métropole  CLOSED_TEMPORARILY   

                                                  types  price_level  \
925   ['restaurant', 'point_of_interest', 'food', 'e...          NaN   
1305  ['restaurant', 'bar', 'food', 'point_of_intere...          NaN   

      ta_price_level  rating  user_ratings_total  ta_rating  \
925              NaN     NaN                 NaN        NaN   
1305             2.0     4.6               114.0        5.0   

      ta_user_ratings_total  combined_rating  geometry.location.lat  \
925                     NaN              NaN              46.203269   
1305                    8.0              4.6              46.202873   

      geometry.location.lng  \
925                6.145108   
1305               6.150983   

                                      formatted_address  zip_code    city  \
925   Rue de la Confédération 8, 1204 Genève, Switze...      1204  Genève   
1305          Rue du Prince 6, 1204 Genève, Switzerland      1204  Genève   

     international_phone_number                          website  \
925                         NaN             http://www.le180.ch/   
1305           +41 22 840 02 02  http://www.bistrotmetropole.ch/   

     google_category ta_4sq_cuisine1 ta_4sq_cuisine2 ta_4sq_cuisine3  \
925                0               0               0               0   
1305      Restaurant             NaN               0               0   

     ta_4sq_cuisine4 ta_4sq_cuisine5 main_category_1 main_category_2  \
925                0               0               0             NaN   
1305               0               0               0             NaN   

     main_category_3 main_category_4 combined_main_category sub_category  \
925              NaN             NaN                    NaN          NaN   
1305             NaN             NaN                    NaN          NaN   

      neighboorhood_area_km2  resident_population_total  population_age_0_14  \
925                     1.06                       8039                  825   
1305                    1.06                       8039                  825   

      population_15_19  population_20_24  population_25_34  population_35_44  \
925                364               528              1416              1277   
1305               364               528              1416              1277   

      population_45_54  population_54_64  population_65+  resident_per_km2  \
925               1249              1070            1310              7554   
1305              1249              1070            1310              7554   

      brut_income_annual_per_person  cluster  
925                          191726        3  
1305                         191726        0

In [54]:
df_rest['combined_main_category'].unique()

array(['European, Swiss', 'General / Restaurant',
       'Italian, French, European', 'European, Portuguese',
       'Swiss, European', 'Japanese', 'Chinese', 'Chicken', 'Italian',
       'Swiss', 'General / Café', 'Pizza', 'Seafood, European, Swiss',
       'Portuguese', 'Pizza, French', 'Steakhouse, Barbecue, Grill',
       'South American', 'Seafood, European', 'Pizza, Italian',
       'Hamburger, American', 'General / Fast food, Snacks, Take Away',
       'Pizza, Italian, European', 'Steakhouse, Barbecue, Grill, Turkish',
       'Italian, Swiss, European', 'General / Bar, Pub, Bistro',
       'French, Swiss, European', 'Lebanese', 'French, European, Swiss',
       'Italian, European, French', 'Turkish', 'Swiss, French, European',
       'French', 'Italian, Pizza', 'Italian, European', 'Indian',
       'Spanish, Steakhouse, Barbecue, Grill',
       'Portuguese, Swiss, European', 'French, Swiss', 'Thai',
       'Other Middle Eastern', 'Mexican', 'Other Asian',
       'Swiss, European

In [66]:
df_categories = pd.DataFrame()
df_categories['restaurant_category'] = df_rest[['main_category_1',
                                               'main_category_2',
                                               'main_category_3',
                                               'main_category_4']]\
                                                .apply(pd.Series.value_counts)\
                                                .sum(axis=1)\
                                                .sort_values(ascending=False)\
                                                .astype(int)
df_categories

restaurant_category
General / Restaurant                                    314
European                                                203
French                                                  191
Italian                                                 166
General / Fast food, Snacks, Take Away                  130
Pizza                                                    94
Swiss                                                    92
General / Bar, Pub, Bistro                               89
Japanese                                                 78
General / Café                                           61
Thai                                                     54
Chinese                                                  52
American                                                 47
Hamburger                                                40
Lebanese                                                 37
Other Asian                                              36
Turkish                                                  36
South American                                           34
Indian                                                   33
Steakhouse, Barbecue, Grill                              30
Mexican                                                  23
Portuguese                                               22
African                                                  21
Seafood                                                  20
Spanish                                                  19
Vegan, Vegetarian, Salad                                 18
Other Middle Eastern                                     18
Chicken                                                  15
Hawaiian                                                 14
All Other                                                14
0                                                         2

In [ ]:
def rest_type(X):
    if X.isin "European":
        return X
    elif X.isin "American":
        return X
    elif X.isin "Asian":
        return X

In [71]:
# contains Asian
Asian=df_rest[df_rest['combined_main_category'].str.contains("Asian", na=False)]
Asian['combined_main_category'].unique()

array(['Other Asian', 'Hamburger, Other Asian',
       'Chinese, Thai, Other Asian',
       'Other Asian, Steakhouse, Barbecue, Grill', 'Chinese, Other Asian'],
      dtype=object)

In [69]:
# contains American

American=df_rest[df_rest['combined_main_category'].str.contains("American", na=False)]
American['combined_main_category'].unique()

array(['South American', 'Hamburger, American',
       'South American, Steakhouse, Barbecue, Grill',
       'American, Swiss, European', 'American, European',
       'General / Fast food, Snacks, Take Away, American, European',
       'European, American, Swiss', 'Seafood, American',
       'Hamburger, Steakhouse, Barbecue, Grill, American',
       'American, Steakhouse, Barbecue, Grill', 'American'], dtype=object)

In [67]:
# contains European

European=df_rest[df_rest['combined_main_category'].str.contains("European", na=False)]
European['combined_main_category'].unique()

array(['European, Swiss', 'Italian, French, European',
       'European, Portuguese', 'Swiss, European',
       'Seafood, European, Swiss', 'Seafood, European',
       'Pizza, Italian, European', 'Italian, Swiss, European',
       'French, Swiss, European', 'French, European, Swiss',
       'Italian, European, French', 'Swiss, French, European',
       'Italian, European', 'Portuguese, Swiss, European',
       'Swiss, European, French', 'Italian, Pizza, European',
       'European, French',
       'European, French, Steakhouse, Barbecue, Grill',
       'Swiss, Italian, French, European', 'Portuguese, European',
       'European, French, Seafood', 'French, European',
       'Spanish, European', 'European',
       'French, European, Portuguese, Italian',
       'Steakhouse, Barbecue, Grill, European, French',
       'French, European, Seafood',
       'Steakhouse, Barbecue, Grill, European',
       'Lebanese, Swiss, European', 'Italian, European, Pizza',
       'European, Turkish', 'Amer

In [9]:
df1=df_rest[['google_category', 'ta_4sq_cuisine1', 'ta_4sq_cuisine2',
       'ta_4sq_cuisine3', 'ta_4sq_cuisine4', 'ta_4sq_cuisine5',
       'main_category_1', 'main_category_2', 'main_category_3',
       'main_category_4', 'combined_main_category', 'sub_category']]

In [13]:
df1_split=df1["google_category"].str.split(" ", expand=True)

In [14]:
df1_split[0].unique()

array(['Restaurant', 'Italian', 'Asian', 'Chicken', 'Swiss', 'Pizza',
       'Portuguese', 'Brazilian', 'Takeout', 'Sushi', 'Bar', 'Japanese',
       'Tea', 'Lebanese', 'Down', 'Colombian', 'Bakery', 'South',
       'Caterer', 'Fast', 'Doner', 'Indian', 'Spanish', 'Thai', 'Turkish',
       'Kebab', 'Taco', 'Chinese', 'Sandwich', 'Brewery', 'Cafe', 'Sri',
       'Stand', 'Eritrean', 'Mexican', 'Irish', 'Argentinian',
       'Indonesian', 'European', 'Vegetarian', 'African', 'Hookah',
       'Pan-Asian', 'Vietnamese', 'Hawaiian', 'French', 'Dim', 'Salad',
       'Noodle', 'Taiwanese', 'Fusion', 'Ramen', 'Hamburger', 'Bistro',
       'Snack', 'Lunch', 'Diner', 'Poke', 'Coffee', 'Brunch', 'Latin',
       'Peruvian', '0', 'Take', 'Eastern', 'Dessert', 'Steak',
       'Ethiopian', 'Liquor', 'Halal', 'Authentic', 'Hawker', 'Crêperie',
       'Moroccan', 'Fine', 'Wok', 'Wine', 'Raw', 'Pretzel', 'Korean',
       'Greek', 'Tex-Mex', 'Health', 'Fruit', 'Pub', 'Filipino',
       'Barbecue', 'Fondu

In [15]:
df1_split[1].unique()

array([None, 'restaurant', 'takeaway', 'Restaurant', '&', 'house', 'home',
       'American', 'food', 'kebab', 'delivery', 'shop', 'Lankan', 'bar',
       'pub', 'sum', 'Away', 'European', 'store', 'Japanese', 'stall',
       'dining', 'fusion', 'parlor', 'tea', 'Muslim', 'French', 'court',
       'Bar', 'Eastern', 'Indian', 'service', 'delicatessen', 'grocery',
       'away', 'dish', 'hall', 'cellar'], dtype=object)

In [16]:
df1_split[2].unique()

array([None, 'grill', 'cooking', 'restaurant', 'store'], dtype=object)

In [17]:
ta4sq_split=df1["ta_4sq_cuisine1"].str.split(" ", expand=True)

In [18]:
ta4sq_split[0].unique()

array(['European', '0', 'Italian', 'Swiss', 'Japanese', 'Chinese',
       'Cafes', 'Seafood', 'General', 'French', 'Fast', 'Beer', 'Café',
       'Lebanese', 'Portuguese', 'Colombian', 'Bakery', 'South', 'Bar',
       'Pizza', 'Brazilian', 'Kebab', 'Indian', 'Steakhouse', 'Turkish',
       'Thai', 'Middle', 'Karaoke', 'International', 'Taco', 'Sushi',
       'Asian', 'Cafe', 'Healthy', 'Mexican', 'Grocery', 'Irish',
       'Spanish', 'African', 'Sandwich', 'Salad', 'Tapas', 'Falafel',
       'American', 'Wine', 'Tea', 'Mediterranean', 'Lounge', 'Hawaiian',
       'Fusion', 'Fried', 'Latin', 'Peruvian', 'Concert', 'Pizzeria',
       'Dining', 'Bistro', 'Gastropub', 'Belgian', 'Diner', nan, 'Deli',
       'Burger', 'Brew', 'Korean', 'Pakistani', 'Barbecue', 'Coffee',
       'Filipino', 'Food', 'Moroccan', 'Soups', 'Northern-Italian',
       'Soup', '', 'Persian', 'Vietnamese', 'Poke', 'Contemporary',
       'Greek', 'Tibetan'], dtype=object)

In [19]:
ta4sq_split[1].unique()

array([None, '/', 'Food', 'Bar', 'American', 'Eastern', 'Store', 'Room',
       'Chicken', 'Hall', 'and', nan, 'Pub', 'Shop', 'Court', 'Truck', '',
       'bars'], dtype=object)

In [20]:
ta4sq_split[2].unique()

array([None, 'Fast', '/', 'Joint', 'Drinking', nan, 'Beverage'],
      dtype=object)

In [22]:
# combined splitted dataaframes for google category and ta4sq

combined=result = pd.concat([df1_split, ta4sq_split], axis=1)
combined

0           1     2     3         0     1     2     3
0     Restaurant        None  None  None  European  None  None  None
1     Restaurant        None  None  None         0  None  None  None
2        Italian  restaurant  None  None   Italian  None  None  None
3     Restaurant        None  None  None         0  None  None  None
4     Restaurant        None  None  None         0  None  None  None
...          ...         ...   ...   ...       ...   ...   ...   ...
1608      Bakery        None  None  None   Italian  None  None  None
1609    American  restaurant  None  None         0  None  None  None
1610       Swiss  restaurant  None  None         0  None  None  None
1611  Restaurant        None  None  None         0  None  None  None
1612  Restaurant        None  None  None         0  None  None  None

[1613 rows x 8 columns]

In [23]:
# manin category list
main_list=['Restaurant','Takeout', 'Sushi', 'Bar',
       'Tea', 'Down','Bakery', 'South',
       'Caterer', 'Fast', 'Sandwich', 'Brewery', 'Cafe', 'Sri',
       'Stand', 'Dim', 'Salad',
       'Noodle', 'Fusion', 'Ramen', 'Hamburger', 'Bistro',
       'Snack', 'Lunch', 'Diner', 'Poke', 'Coffee', 'Brunch',
       'Take', 'Eastern', 'Dessert', 'Steak',
       'Ethiopian', 'Liquor', 'Halal', 'Authentic', 'Hawker', 'Crêperie',
     'Fine', 'Wok', 'Wine', 'Raw', 'Pretzel', 
       'Tex-Mex', 'Health', 'Fruit', 'Pub',
       'Barbecue', 'Fondue', 'Meal', 'Bubble', 'Lounge',
        'Cafeteria', 'Haute', 'Food', 'Juice',
       'Cocktail', 'Buffet', 'Chophouse', 'Okonomiyaki', 'Middle',
        'Modern', 'Self', 'Brasserie',
       'Seafood', 'Tapas', 'Grill', 'General',
       'Organic', 'Deli', 'Fish', 'Meat', 'Beer', 'Oyster', 'Gourmet',
       'Mandarin', 'Family', 'Traditional', 'Delivery', 'restaurant', 'takeaway', 'Restaurant', '&', 'house', 'home',
       'food', 'delivery', 'shop', 'bar',
       'pub', 'sum', 'Away', 'store', 'stall',
       'dining', 'fusion', 'parlor', 'tea', 'court',
       'Bar','service', 'delicatessen', 'grocery',
       'away', 'dish', 'hall', 'cellar', 'grill', 'cooking', 'store', 'Cafes', 'Seafood', 'General','Fast', 'Beer', 'Café',
          'Bakery', 'South', 'Bar','International', 'Taco','Food', 'Store', 'Room',
       'Chicken', 'Hall', 'and','Pub', 'Shop', 'Court', 'Truck','Beverage',
       'bars']

In [ ]:
#unclassified terms
non_classified=[" Down"]

In [30]:
# find in df_rest restaurants and then subst


restaurant=df_rest[df_rest['google_category'].isin(["restaurant", "Restaurant"])]
len(restaurant)

782

for 782 restaurants, to further classify we will look into column of ta_4sq_cuisine1, 
because the column google_category contains only entry of " restaurant" 

In [32]:
restaurant["google_category"].str.split(" ", expand=True)

0
0     Restaurant
1     Restaurant
3     Restaurant
4     Restaurant
5     Restaurant
...          ...
1605  Restaurant
1606  Restaurant
1607  Restaurant
1611  Restaurant
1612  Restaurant

[782 rows x 1 columns]

In [34]:
rest_restaurant=df_rest[~df_rest['google_category'].isin(["restaurant", "Restaurant"])]
len(rest_restaurant)

831

In [38]:
rest_restaurant['google_category'].unique()

array(['Italian restaurant', 'Asian restaurant', 'Chicken restaurant',
       'Swiss restaurant', 'Pizza takeaway', 'Portuguese restaurant',
       'Pizza restaurant', 'Brazilian restaurant', 'Takeout Restaurant',
       'Sushi restaurant', 'Bar & grill', 'Japanese restaurant',
       'Tea house', 'Lebanese restaurant', 'Down home cooking restaurant',
       'Colombian restaurant', 'Bakery', 'South American restaurant',
       'Bar', 'Caterer', 'Fast food restaurant', 'Doner kebab restaurant',
       'Indian restaurant', 'Spanish restaurant', 'Pizza delivery',
       'Thai restaurant', 'Turkish restaurant', 'Kebab shop',
       'Taco restaurant', 'Chinese restaurant', 'Sandwich shop',
       'Brewery', 'Cafe', 'Takeout restaurant', 'Sri Lankan restaurant',
       'Stand bar', 'Eritrean restaurant', 'Mexican restaurant',
       'Irish pub', 'Argentinian restaurant', 'Indonesian restaurant',
       'Indian takeaway', 'European restaurant', 'Vegetarian restaurant',
       'African restaur

In [50]:
contain_restaurant=rest_restaurant[rest_restaurant['google_category'].str.contains("restaurant")]
len(contain_restaurant)

580

In [52]:
others=rest_restaurant[~rest_restaurant['google_category'].str.contains("restaurant")]
others['google_category'].unique()

array(['Pizza takeaway', 'Takeout Restaurant', 'Bar & grill', 'Tea house',
       'Bakery', 'Bar', 'Caterer', 'Pizza delivery', 'Kebab shop',
       'Sandwich shop', 'Brewery', 'Cafe', 'Stand bar', 'Irish pub',
       'Indian takeaway', 'Hookah bar', 'Salad shop', 'Noodle shop',
       'Bistro', 'Snack bar', 'Diner', 'Poke bar', 'Coffee shop', '0',
       'Take Away', 'Dessert shop', 'Steak house', 'Liquor store',
       'Hawker stall', 'Crêperie', 'Wine bar', 'Pretzel store',
       'Fruit parlor', 'Pub', 'Meal delivery', 'Bubble tea store',
       'Lounge', 'Sushi takeaway', 'Cafeteria', 'Pizza', 'Food court',
       'Juice', 'Cocktail Bar', 'Brasserie', 'Japanese delicatessen',
       'Cocktail bar', 'Lebanese', 'Tapas bar', 'Italian grocery store',
       'Grill', 'General store', 'Deli', 'Take away', 'Fish store',
       'Dim sum', 'Beer hall', 'Gourmet grocery store', 'Wine cellar',
       'Juice shop', 'Delivery Restaurant'], dtype=object)

In [53]:
len(others)

251

In [ ]:

Goegraphy(Asian, European,Amerikan)
takeaway
Bar
Cockail Bar
Brasserie
Cafe
FastFood
Pub
Seafood
Grill



In [40]:
rest_restaurant[rest_restaurant['google_category'].str.contains("restaurant")]

place_id  tripadvisor_id                    fsq_id  \
2     ChIJPcYsG_BljEcR4kTFqBJJoK4       6720272.0                       NaN   
11    ChIJL1Sw0ZdkjEcRDN1B5HLWcAI      10123088.0  565da093498e48bf87ebb886   
13    ChIJN97gX2NljEcRfnSAXk9QFyE             NaN                       NaN   
15    ChIJFw7PgDV7jEcR8KRa2PoMgX8             NaN                       NaN   
21    ChIJH-hMPchkjEcR9ogF6QK8TbY             NaN  f407026f4281417f20aec48f   
...                           ...             ...                       ...   
1602  ChIJweODnk9ljEcReOxhBKacdR0             NaN                       NaN   
1603  ChIJ8ShOF0lljEcR_TWOQyz8l9Y      10713054.0                       NaN   
1604  ChIJpYWlTUlljEcR1wQr6MUxT4M      13335017.0                       NaN   
1609  ChIJHSzAC1JljEcRjr7q009gZe4        784992.0                       NaN   
1610  ChIJN-0qEVJljEcR34z73PN9zUU             NaN  5389d742498e46ffbbc28d5c   

                                                   name business_status  \
2                                         Le Versailles     OPERATIONAL   
11                                       AKIKO Balexert     OPERATIONAL   
13                                           Max Poulet     OPERATIONAL   
15                                  Café de la Terrasse     OPERATIONAL   
21    Association Démocratique des Travailleurs Port...     OPERATIONAL   
...                                                 ...             ...   
1602                                          Greendy's     OPERATIONAL   
1603                                          La Storia     OPERATIONAL   
1604                                       KATANA Sushi     OPERATIONAL   
1609                                        Road Runner     OPERATIONAL   
1610  La courbe du goût, café restaurant de Grange-C...     OPERATIONAL   

                                                  types  price_level  \
2     ['restaurant', 'point_of_interest', 'food', 'e...          NaN   
11    ['cafe', 'meal_takeaway', 'restaurant', 'food'...          2.0   
13    ['restaurant', 'food', 'point_of_interest', 'e...          NaN   
15    ['restaurant', 'food', 'point_of_interest', 'e...          2.0   
21    ['restaurant', 'point_of_interest', 'food', 'e...          NaN   
...                                                 ...          ...   
1602  ['restaurant', 'food', 'point_of_interest', 'e...          NaN   
1603  ['meal_takeaway', 'restaurant', 'point_of_inte...          2.0   
1604  ['restaurant', 'food', 'point_of_interest', 'e...          NaN   
1609  ['restaurant', 'food', 'point_of_interest', 'e...          2.0   
1610  ['restaurant', 'food', 'point_of_interest', 'e...          2.0   

      ta_price_level  rating  user_ratings_total  ...  population_15_19  \
2                2.0     3.9                56.0  ...              1547   
11               2.0     3.4               251.0  ...               825   
13               NaN     NaN                 NaN  ...              1547   
15               NaN     4.3               199.0  ...              1956   
21               NaN     4.5                23.0  ...              1547   
...              ...     ...                 ...  ...               ...   
1602             NaN     4.7                20.0  ...               780   
1603             2.0     4.3               235.0  ...               780   
1604             2.0     4.6               161.0  ...               713   
1609             2.0     4.3               376.0  ...               713   
1610             NaN     4.3               250.0  ...               746   

      population_20_24  population_25_34  population_35_44  population_45_54  \
2                 1937              5163              5549              4924   
11                 839              2130              2701              2370   
13                1937              5163              5549              4924   
15                2148              4721              5127              5017   
21       

In [ ]:
main_category=["Restaurant", "takeaway", " Bar & grill", 'Tea house','Bakery','Bar', 'Caterer','Fast food restaurant',
              'Pizza', 'Sandwich shop','Brewery', 'Cafe','Hookah bar','Salad shop','Noodle shop','Bistro', 'Snack bar',
       'Lunch restaurant', 'Diner', 'Poke bar', 'Coffee shop','Brunch restaurant','Dessert shop', 'Steak house','Liquor store',
              'Hawker stall', 'Crêperie','Fine dining restaurant','Wok restaurant', 'Wine bar',
       'Raw food restaurant', 'Pretzel store','Tex-Mex restaurant','Fruit parlor', 'Pub', 'Meal delivery',
       'Bubble tea store','Lounge','Cafeteria',
       'Pizza', 'Food court', 'Juice', 'Cocktail Bar',
       'Buffet restaurant', 'Chophouse restaurant',
       'Okonomiyaki restaurant','Self service restaurant', 'Brasserie',
       'Japanese delicatessen', 'Seafood restaurant', 'Cocktail bar',
       'Lebanese', 'Tapas bar', 'Italian grocery store',
        'Grill', 'General store', 
       'Organic restaurant', 'Deli', 'Take away', 'Fish store', 'Dim sum',
       'Meat dish restaurant', 'Beer hall', 'Oyster bar restaurant',
       'Gourmet grocery store', 'Wine cellar',
       'Juice shop', 'Family restaurant', 'Traditional restaurant',
       'Tapas restaurant', 'Delivery Restaurant']

In [ ]:
Restaurants=

In [ ]:
wegan=

In [46]:
Take_aways=rest_restaurant[rest_restaurant['google_category'].str.contains("takeaway")]
Take_aways

place_id  tripadvisor_id fsq_id  \
17    ChIJI05JTr5kjEcRTffOWTQzCNE             NaN    NaN   
18    ChIJNVpwC79kjEcRqKfKmxCTkgE             NaN    NaN   
196   ChIJp_pTqmZljEcRBNxhT_Ux26w      23305969.0    NaN   
232   ChIJYeAnX9NkjEcRoP0ibig0G9U             NaN    NaN   
310   ChIJTYve55dljEcRdFv8gSA4pHc             NaN    NaN   
864   ChIJX87z8DZljEcROjf9paJTQI0             NaN    NaN   
1014  ChIJCQO6g6ZljEcR2BRFEg1ICvU             NaN    NaN   
1100  ChIJETbrCkFljEcRMCYiN3L_UQU      21329381.0    NaN   
1395  ChIJiTfZhHNljEcRaxC60DmcztI             NaN    NaN   
1495  ChIJIS3AAe9ljEcReleRiiW-kdY             NaN    NaN   

                                    name     business_status  \
17                            Formule 10         OPERATIONAL   
18                      Au Siciliano DOC         OPERATIONAL   
196                   Delhi Tadka Geneve         OPERATIONAL   
232                            Pizza mia         OPERATIONAL   
310                          Açaí Genève         OPERATIONAL   
864                           Hona Sushi  CLOSED_TEMPORARILY   
1014  Temaki Bar - Bongénie (6ème étage)         OPERATIONAL   
1100     Temaki Bar - Quai du Mont-Blanc         OPERATIONAL   
1395                  Pizzeria RiverSide         OPERATIONAL   
1495             Pizzeria des Eaux-Vives         OPERATIONAL   

                                                  types  price_level  \
17    ['meal_takeaway', 'restaurant', 'food', 'point...          1.0   
18    ['meal_takeaway', 'bar', 'restaurant', 'food',...          NaN   
196   ['meal_takeaway', 'restaurant', 'point_of_inte...          NaN   
232   ['meal_takeaway', 'restaurant', 'food', 'point...          NaN   
310   ['meal_takeaway', 'restaurant', 'food', 'point...          NaN   
864   ['meal_takeaway', 'restaurant', 'food', 'point...          NaN   
1014  ['meal_takeaway', 'restaurant', 'food', 'point...          NaN   
1100  ['meal_takeaway', 'restaurant', 'food', 'point...          NaN   
1395  ['meal_takeaway', 'restaurant', 'food', 'point...          NaN   
1495  ['meal_takeaway', 'restaurant', 'point_of_inte...          NaN   

      ta_price_level  rating  user_ratings_total  ...  population_15_19  \
17               NaN     4.2               101.0  ...               825   
18               NaN     NaN                 NaN  ...              1547   
196              2.0     4.9                14.0  ...               879   
232              NaN     4.5                15.0  ...              1264   
310              NaN     NaN                 NaN  ...              1156   
864              NaN     4.8                16.0  ...              1156   
1014             NaN     NaN                 NaN  ...               364   
1100             3.0     4.2                59.0  ...              1156   
1395             NaN     4.8                10.0  ...               780   
1495             NaN     5.0                37.0  ...               780   

      population_20_24  population_25_34  population_35_44  population_45_54  \
17                 839              2130              2701              2370   
18                1937              5163              5549              4924   
196               1356              3501              3555              3028   
232               2253              7285              5620              4812   
310               1611              5621              5389              4675   
864               1611              5621              5389              4675   
1014               528              1416              1277              1249   
1100              1611              5621              5389              4675   
1395               996              3839              3699              3048   
1495               996              3839              3699              3048   

     population_54_64  population_65+ resident_per_km2  \
17               2067            3313            10625   
18               4290            5625      

In [43]:
pizza=rest_restaurant[rest_restaurant['google_category'].str.contains("Pizza")]
pizza

place_id  tripadvisor_id                    fsq_id  \
17    ChIJI05JTr5kjEcRTffOWTQzCNE             NaN                       NaN   
18    ChIJNVpwC79kjEcRqKfKmxCTkgE             NaN                       NaN   
24    ChIJCyhiu8BkjEcRRRNmqkEJ8g0      11963186.0                       NaN   
31    ChIJZ0HOdJtljEcR8x0Gn7jTlXo       9804656.0  151198066d2c4f4ee6c65730   
36    ChIJdRFop8hkjEcRjhEtkUr_tJc       2491589.0  4da88bf1fa8c4175d0bcb33f   
68    ChIJWeF-pMNkjEcRCKE9maeYDqg      17458201.0                       NaN   
91    ChIJWeF-pMNkjEcRFM0-x2UqV1A             NaN                       NaN   
225   ChIJu4-hjSh7jEcRgATIixpjDWI             NaN                       NaN   
232   ChIJYeAnX9NkjEcRoP0ibig0G9U             NaN                       NaN   
254   ChIJg05HvVpljEcRoXIcdU3VwwY             NaN                       NaN   
265   ChIJazmoFEF7jEcR5b0fKNZg5-Y      23257523.0                       NaN   
277   ChIJ0W8TTmNljEcRsb82xXhXvfE             NaN  5935b74aff19025f4f388b7e   
310   ChIJTYve55dljEcRdFv8gSA4pHc             NaN                       NaN   
345   ChIJtzDLCJdljEcR3tLQ0txFeM8      21014811.0                       NaN   
347   ChIJy6O-DOlljEcRXRQPc5dgOrU             NaN                       NaN   
357   ChIJkaAI3Sl7jEcRtS3AcbsGtDY      14781549.0  4f453f08e4b0b5bea5453100   
399   ChIJR-4V991kjEcRqV3c5R70Tdk        813469.0  4ff3440fe4b0fecb241ffdd8   
454   ChIJVz28xdB6jEcRQRgp0Lwt19c             NaN  5ac3828035f9833cb96b4a29   
486   ChIJbetEgNdkjEcRf4Y9MQ4CPBE      20052160.0  0384f779b76b42ec4c5fc274   
635   ChIJ21nu8CdljEcRTcdOHbXBCtU        785033.0  4adcdab4f964a520665021e3   
639   ChIJyeg7h9hkjEcRvbaoLxTqwZk             NaN                       NaN   
679   ChIJxa3APdR6jEcR9UGoRrQiH9M       4364360.0                       NaN   
837   ChIJBTQhYSlljEcR-b4ePdh5Moc       7270056.0  4e9eb102e5fa0c382e117b86   
850   ChIJ0Q6dJyZljEcR9X_HKNsvhA8       3650156.0  526fa1d6498ee1d17a8b2aec   
920   ChIJ29kiCSxljEcR8gOniUAofwk             NaN  4bc871c16501c9b6a2164029   
1011  ChIJ0z6WgtJljEcRCzBbfFWQJJI       7002847.0  5b3774b5c0cacb002c09423f   
1090  ChIJrynelC5ljEcRNVVe30wX_wc       1319335.0  4b4e3050f964a52091e426e3   
1108  ChIJjx9Vq69ljEcRN5sjCvqX-D0             NaN                       NaN   
1186  ChIJyX8yCSRljEcRDe5p3amZCEA        805814.0                       NaN   
1362  ChIJ09yVkMx6jEcR6wYgvNRC_WY       7885610.0  547981949df94a8c3d01450f   
1386  ChIJe_HevzNljEcRbiVBC43V2x4       8467786.0  557ea993498edbe4e131e981   
1395  ChIJiTfZhHNljEcRaxC60DmcztI             NaN                       NaN   
1419  ChIJYSFo8TNljEcRrRMm9hpAudg      10068743.0  573c33fd498e6f78c3d91fd4   
1437  ChIJB3MxxTBljEcRduCOecxA12c        776629.0  4b311960f964a520330025e3   
1489  ChIJFw5R1zZljEcRddG0NtTSF90        784995.0  4c339005a0ced13a01d7166e   
1495  ChIJIS3AAe9ljEcReleRiiW-kdY             NaN                       NaN   
1536  ChIJL8uA74JljEcR30n-G3iWmaA       7018443.0                       NaN   
1540  ChIJbdpAczdljEcRa403BKZexjY       7142904.0  4bc6f46d14d79521fb7666e9   
1567  ChIJpVwdfRZljEcRtlj4HQG9Yjw             NaN                       NaN   
1571  ChIJA4CmC0pljEcR4euPABLOTRU        810437.0                       NaN   
1601  ChIJ28Joch9ljEcRD7SvpICBvkc      15043241.0                       NaN   

                                            name     business_status  \
17                                    Formule 10         OPERATIONAL   
18                              Au Siciliano DOC         OPERATIONAL   
24                        Restaurant La Fontaine         OPERATIONAL   
31                                  Cafe Salento         OPERATIONAL   
36                                 Da Pulcinella         OPERATIONAL   
68                                      Domino's         OPERATIONAL   
91                        Rolling Pizza Servette         OPERATIONAL   
225                                     La Ruota         OPERATIONAL   
232                                    Pizza mia         O

In [44]:
pizza['google_category'].unique()

array(['Pizza takeaway', 'Pizza restaurant', 'Pizza delivery', 'Pizza'],
      dtype=object)

In [ ]:
left_overs=['Down home cooking restaurant','Dim sum restaurant','Food court',]

## convert to restaurant

convert_rest=["Kebab shop" ]

## Burak's code

In [5]:
tmp = df_rest[df_rest['price_level'] > 0]
len(tmp)

592

In [6]:
tmp = df_rest[(df_rest['price_level'] > 0) | (df_rest['ta_price_level'] > 0)]
len(tmp)

910

In [7]:
df_rest.isnull().sum() / len(df_rest) * 100

place_id                          0.000000
tripadvisor_id                   40.545567
fsq_id                           58.028518
name                              0.000000
business_status                   0.000000
types                             0.000000
price_level                      63.298202
ta_price_level                   50.960942
rating                           13.639182
user_ratings_total               13.639182
ta_rating                        44.451333
ta_user_ratings_total            44.451333
combined_rating                  12.213267
geometry.location.lat             0.000000
geometry.location.lng             0.000000
formatted_address                 0.000000
zip_code                          0.000000
city                              0.000000
international_phone_number       13.577185
website                          59.702418
google_category                   0.000000
ta_4sq_cuisine1                   0.371978
ta_4sq_cuisine2                   4.091754
ta_4sq_cuis

In [103]:
# Finding total categories
df_categories = pd.DataFrame()
df_categories['restaurant_category'] = df_rest[['main_category_1',
                                               'main_category_2',
                                               'main_category_3',
                                               'main_category_4']]\
                                                .apply(pd.Series.value_counts)\
                                                .sum(axis=1)\
                                                .sort_values(ascending=False)\
                                                .astype(int)
list_categories = df_categories.index.to_list()

In [117]:
list_categories

['General / Restaurant',
 'European',
 'French',
 'Italian',
 'General / Fast food / Snacks / Take Away',
 'Pizza',
 'Swiss',
 'General / Bar / Pub / Bistro',
 'Japanese',
 'General / Café',
 'Thai',
 'Chinese',
 'American',
 'Hamburger',
 'Lebanese',
 'Other Asian',
 'Turkish',
 'South American',
 'Indian',
 'Steakhouse / Barbecue / Grill',
 'Mexican',
 'Portuguese',
 'African',
 'Seafood',
 'Spanish',
 'Vegan / Vegetarian / Salad',
 'Other Middle Eastern',
 'Chicken',
 'Hawaiian',
 'All Other',
 '0']

In [105]:
df_regression = df_rest[(df_rest['price_level'] > 0) & (df_rest['combined_rating'] > 0) & (df_rest['combined_main_category'] != 0)]\
    [['geometry.location.lat','geometry.location.lng','price_level','combined_main_category','combined_rating']]
df_regression

geometry.location.lat  geometry.location.lng  price_level  \
0                 46.205232               6.114945          2.0   
3                 46.196924               6.115081          2.0   
6                 46.210396               6.114510          2.0   
7                 46.210587               6.113854          2.0   
9                 46.220396               6.114030          2.0   
...                     ...                    ...          ...   
1606              46.207746               6.169275          3.0   
1608              46.192811               6.169873          2.0   
1609              46.199892               6.174117          2.0   
1610              46.199773               6.174561          2.0   
1612              46.201690               6.174614          1.0   

                        combined_main_category  combined_rating  
0                              European, Swiss              4.1  
3                         General / Restaurant              4.5  
6                         General / Restaurant              3.9  
7                         General / Restaurant              4.3  
9                              Swiss, European              4.1  
...                                        ...              ...  
1606                      General / Restaurant              4.4  
1608  General / Fast food / Snacks / Take Away              4.0  
1609                                  American              4.1  
1610                                     Swiss              4.3  
1612                      General / Restaurant              4.3  

[592 rows x 5 columns]

In [120]:
list_categories[0:-1]

['General / Restaurant',
 'European',
 'French',
 'Italian',
 'General / Fast food / Snacks / Take Away',
 'Pizza',
 'Swiss',
 'General / Bar / Pub / Bistro',
 'Japanese',
 'General / Café',
 'Thai',
 'Chinese',
 'American',
 'Hamburger',
 'Lebanese',
 'Other Asian',
 'Turkish',
 'South American',
 'Indian',
 'Steakhouse / Barbecue / Grill',
 'Mexican',
 'Portuguese',
 'African',
 'Seafood',
 'Spanish',
 'Vegan / Vegetarian / Salad',
 'Other Middle Eastern',
 'Chicken',
 'Hawaiian',
 'All Other']

In [121]:
tmpx = pd.DataFrame()
tmpx['main_category'] = df_regression['combined_main_category']
for rest_type in list_categories[0:-1]:
    tmpx[rest_type] = 0

In [122]:
tmpx = tmpx.reset_index(drop=True)

In [123]:
tmpx

main_category  General / Restaurant  European  \
0                             European, Swiss                     0         0   
1                        General / Restaurant                     0         0   
2                        General / Restaurant                     0         0   
3                        General / Restaurant                     0         0   
4                             Swiss, European                     0         0   
..                                        ...                   ...       ...   
587                      General / Restaurant                     0         0   
588  General / Fast food / Snacks / Take Away                     0         0   
589                                  American                     0         0   
590                                     Swiss                     0         0   
591                      General / Restaurant                     0         0   

     French  Italian  General / Fast food / Snacks / Take Away  Pizza  Swiss  \
0         0        0                                         0      0      0   
1         0        0                                         0      0      0   
2         0        0                                         0      0      0   
3         0        0                                         0      0      0   
4         0        0                                         0      0      0   
..      ...      ...                                       ...    ...    ...   
587       0        0                                         0      0      0   
588       0        0                                         0      0      0   
589       0        0                                         0      0      0   
590       0        0                                         0      0      0   
591       0        0                                         0      0      0   

     General / Bar / Pub / Bistro  Japanese  ...  Mexican  Portuguese  \
0                               0         0  ...        0           0   
1                               0         0  ...        0           0   
2                               0         0  ...        0           0   
3                               0         0  ...        0           0   
4                               0         0  ...        0           0   
..                            ...       ...  ...      ...         ...   
587                             0         0  ...        0           0   
588                             0         0  ...        0           0   
589                             0         0  ...        0           0   
590                             0         0  ...        0           0   
591                             0         0  ...        0           0   

     African  Seafood  Spanish  Vegan / Vegetarian / Salad  \
0          0        0        0                           0   
1          0        0        0                           0   
2          0        0        0                           0   
3          0        0        0                           0   
4          0        0        0                           0   
..       ...      ...      ...                         ...   
587        0        0        0                           0   
588        0        0        0                           0   
589        0        0        0                           0   
590        0        0        0                           0   
591        0        0        0                           0   

     Other Middle Eastern  Chicken  Hawaiian  All Other  
0                       0        0         0          0  
1                       0        0         0          0  
2                       0        0         0          0  
3                       0        0         0          0  
4                       0        0         0          0  
..                    ...      ...       ...        ...  
587                     0        0         0          0  
588         

In [110]:
cat_row = tmpx.iloc[0]['main_category'].split(', ')
cat_row

['European', 'Swiss']

In [111]:
tmpx.loc[0, cat_row] = 1

In [124]:
tmpx.loc[0]

main_category                               European, Swiss
General / Restaurant                                      0
European                                                  0
French                                                    0
Italian                                                   0
General / Fast food / Snacks / Take Away                  0
Pizza                                                     0
Swiss                                                     0
General / Bar / Pub / Bistro                              0
Japanese                                                  0
General / Café                                            0
Thai                                                      0
Chinese                                                   0
American                                                  0
Hamburger                                                 0
Lebanese                                                  0
Other Asian                             

In [125]:
tmpx.loc[0, 'main_category']

'European, Swiss'

In [126]:
for idx in range(len(tmpx)):
    cat_row = tmpx.loc[idx, 'main_category'].split(", ")
    tmpx.loc[idx, cat_row] = 1

In [127]:
tmpx

main_category  General / Restaurant  European  \
0                             European, Swiss                     0         1   
1                        General / Restaurant                     1         0   
2                        General / Restaurant                     1         0   
3                        General / Restaurant                     1         0   
4                             Swiss, European                     0         1   
..                                        ...                   ...       ...   
587                      General / Restaurant                     1         0   
588  General / Fast food / Snacks / Take Away                     0         0   
589                                  American                     0         0   
590                                     Swiss                     0         0   
591                      General / Restaurant                     1         0   

     French  Italian  General / Fast food / Snacks / Take Away  Pizza  Swiss  \
0         0        0                                         0      0      1   
1         0        0                                         0      0      0   
2         0        0                                         0      0      0   
3         0        0                                         0      0      0   
4         0        0                                         0      0      1   
..      ...      ...                                       ...    ...    ...   
587       0        0                                         0      0      0   
588       0        0                                         1      0      0   
589       0        0                                         0      0      0   
590       0        0                                         0      0      1   
591       0        0                                         0      0      0   

     General / Bar / Pub / Bistro  Japanese  ...  Mexican  Portuguese  \
0                               0         0  ...        0           0   
1                               0         0  ...        0           0   
2                               0         0  ...        0           0   
3                               0         0  ...        0           0   
4                               0         0  ...        0           0   
..                            ...       ...  ...      ...         ...   
587                             0         0  ...        0           0   
588                             0         0  ...        0           0   
589                             0         0  ...        0           0   
590                             0         0  ...        0           0   
591                             0         0  ...        0           0   

     African  Seafood  Spanish  Vegan / Vegetarian / Salad  \
0          0        0        0                           0   
1          0        0        0                           0   
2          0        0        0                           0   
3          0        0        0                           0   
4          0        0        0                           0   
..       ...      ...      ...                         ...   
587        0        0        0                           0   
588        0        0        0                           0   
589        0        0        0                           0   
590        0        0        0                           0   
591        0        0        0                           0   

     Other Middle Eastern  Chicken  Hawaiian  All Other  
0                       0        0         0          0  
1                       0        0         0          0  
2                       0        0         0          0  
3                       0        0         0          0  
4                       0        0         0          0  
..                    ...      ...       ...        ...  
587                     0        0         0          0  
588         

# Creating Restaurant Category Clusters

In [130]:
# creating a dictionary for limiting restaurant categories

rest_keys = [
    'General / Restaurant',
    'European',
    'French',
    'Italian',
    'General / Fast food / Snacks / Take Away',
    'Pizza',
    'Swiss',
    'General / Bar / Pub / Bistro',
    'Japanese',
    'General / Café',
    'Thai',
    'Chinese',
    'American',
    'Hamburger',
    'Lebanese',
    'Other Asian',
    'Turkish',
    'South American',
    'Indian',
    'Steakhouse / Barbecue / Grill',
    'Mexican',
    'Portuguese',
    'African',
    'Seafood',
    'Spanish',
    'Vegan / Vegetarian / Salad',
    'Other Middle Eastern',
    'Chicken',
    'Hawaiian',
    'All Other']

rest_values = [
    'General',
    'European',
    'European',
    'European',
    'General',
    'Pizza',
    'European',
    'General',
    'Asian',
    'General',
    'Asian',
    'Asian',
    'American',
    'Hamburger',
    'Middle Eastern',
    'Asian',
    'Middle Eastern',
    'South American',
    'Indian',
    'Steakhouse / Barbecue / Grill',
    'Mexican',
    'European',
    'African',
    'Seafood',
    'European',
    'Vegan / Vegetarian / Salad',
    'Middle Eastern',
    'Chicken',
    'Hawaiian',
    'All Other']

rest_dict = {rest_keys[i]: rest_values[i] for i in range(len(rest_keys))}
rest_dict

{'General / Restaurant': 'General',
 'European': 'European',
 'French': 'European',
 'Italian': 'European',
 'General / Fast food / Snacks / Take Away': 'General',
 'Pizza': 'Pizza',
 'Swiss': 'European',
 'General / Bar / Pub / Bistro': 'General',
 'Japanese': 'Asian',
 'General / Café': 'General',
 'Thai': 'Asian',
 'Chinese': 'Asian',
 'American': 'American',
 'Hamburger': 'Hamburger',
 'Lebanese': 'Middle Eastern',
 'Other Asian': 'Asian',
 'Turkish': 'Middle Eastern',
 'South American': 'South American',
 'Indian': 'Indian',
 'Steakhouse / Barbecue / Grill': 'Steakhouse / Barbecue / Grill',
 'Mexican': 'Mexican',
 'Portuguese': 'European',
 'African': 'African',
 'Seafood': 'Seafood',
 'Spanish': 'European',
 'Vegan / Vegetarian / Salad': 'Vegan / Vegetarian / Salad',
 'Other Middle Eastern': 'Middle Eastern',
 'Chicken': 'Chicken',
 'Hawaiian': 'Hawaiian',
 'All Other': 'All Other'}

In [133]:
set(rest_dict.values())

17

In [152]:
tmpy = pd.DataFrame()
tmpy['main_category'] = df_regression['combined_main_category']
for rest_type in set(rest_dict.values()):
    tmpy[rest_type] = 0
tmpy.reset_index(drop=True, inplace=True)

In [153]:
for idx in range(len(tmpy)):
    cat_row = tmpy.loc[idx, 'main_category'].split(", ")
    for cat in cat_row:
        tmpy.loc[idx, rest_dict[cat]] = 1

In [154]:
tmpy

main_category  Seafood  \
0                             European, Swiss        0   
1                        General / Restaurant        0   
2                        General / Restaurant        0   
3                        General / Restaurant        0   
4                             Swiss, European        0   
..                                        ...      ...   
587                      General / Restaurant        0   
588  General / Fast food / Snacks / Take Away        0   
589                                  American        0   
590                                     Swiss        0   
591                      General / Restaurant        0   

     Vegan / Vegetarian / Salad  Middle Eastern  \
0                             0               0   
1                             0               0   
2                             0               0   
3                             0               0   
4                             0               0   
..                          ...             ...   
587                           0               0   
588                           0               0   
589                           0               0   
590                           0               0   
591                           0               0   

     Steakhouse / Barbecue / Grill  Hawaiian  General  American  Indian  \
0                                0         0        0         0       0   
1                                0         0        1         0       0   
2                                0         0        1         0       0   
3                                0         0        1         0       0   
4                                0         0        0         0       0   
..                             ...       ...      ...       ...     ...   
587                              0         0        1         0       0   
588                              0         0        1         0       0   
589                              0         0        0         1       0   
590                              0         0        0         0       0   
591                              0         0        1         0       0   

     European  Hamburger  African  Asian  Pizza  South American  Chicken  \
0           1          0        0      0      0               0        0   
1           0          0        0      0      0               0        0   
2           0          0        0      0      0               0        0   
3           0          0        0      0      0               0        0   
4           1          0        0      0      0               0        0   
..        ...        ...      ...    ...    ...             ...      ...   
587         0          0        0      0      0               0        0   
588         0          0        0      0      0               0        0   
589         0          0        0      0      0               0        0   
590         1          0        0      0      0               0        0   
591         0          0        0      0      0               0        0   

     All Other  Mexican  
0            0        0  
1            0        0  
2            0        0  
3            0        0  
4            0        0  
..         ...      ...  
587          0        0  
588          0        0  
589          0        0  
590          0        0  
591          0        0  

[592 rows x 18 columns]

# Running Regression Analysis

In [209]:
df_regression = df_rest[(df_rest['price_level'] > 0) & (df_rest['combined_rating'] > 0) & (df_rest['combined_main_category'] != 0)]\
    [['geometry.location.lat','geometry.location.lng','price_level','combined_main_category','combined_rating']].reset_index(drop=True)
df_regression.head()

geometry.location.lat  geometry.location.lng  price_level  \
0              46.205232               6.114945          2.0   
1              46.196924               6.115081          2.0   
2              46.210396               6.114510          2.0   
3              46.210587               6.113854          2.0   
4              46.220396               6.114030          2.0   

  combined_main_category  combined_rating  
0        European, Swiss              4.1  
1   General / Restaurant              4.5  
2   General / Restaurant              3.9  
3   General / Restaurant              4.3  
4        Swiss, European              4.1

In [210]:
for rest_type in set(rest_dict.values()):
    df_regression[rest_type] = 0

for idx in range(len(df_regression)):
    row_category = df_regression.loc[idx, 'combined_main_category'].split(", ")
    for category in row_category:
        df_regression.loc[idx, rest_dict[category]] = 1
        
df_regression.drop(columns=['combined_main_category'], inplace=True)

In [211]:
from sklearn.preprocessing import MinMaxScaler

# Instanciate MinMax Scaler 
mm_scaler = MinMaxScaler()

# select features and fit
features = [
    'geometry.location.lat',
    'geometry.location.lng']

mm_scaler.fit(df_regression[features])

# scale
df_regression[features] = mm_scaler.transform(df_regression[features])

In [206]:
from sklearn.preprocessing import OneHotEncoder

ohe = OneHotEncoder(sparse=False) # Instantiate encoder

ohe.fit(df_regression[['price_level']]) # Fit encoder
encoded = ohe.transform(df_regression[['price_level']]) # Encode / Transform GarageFinish
print(encoded)
print(ohe.categories_)

[[0. 1. 0. 0.]
 [0. 1. 0. 0.]
 [0. 1. 0. 0.]
 ...
 [0. 1. 0. 0.]
 [0. 1. 0. 0.]
 [1. 0. 0. 0.]]
[array([1., 2., 3., 4.])]


In [212]:
df_regression['price_level_1'], df_regression['price_level_2'], df_regression['price_level_3'], df_regression['price_level_4'] = encoded.T
df_regression.head()

geometry.location.lat  geometry.location.lng  price_level  combined_rating  \
0               0.484261               0.019853          2.0              4.1   
1               0.307162               0.022085          2.0              4.5   
2               0.594350               0.012711          2.0              3.9   
3               0.598420               0.001937          2.0              4.3   
4               0.807508               0.004836          2.0              4.1   

   Seafood  Vegan / Vegetarian / Salad  Middle Eastern  \
0        0                           0               0   
1        0                           0               0   
2        0                           0               0   
3        0                           0               0   
4        0                           0               0   

   Steakhouse / Barbecue / Grill  Hawaiian  General  ...  Asian  Pizza  \
0                              0         0        0  ...      0      0   
1                              0         0        1  ...      0      0   
2                              0         0        1  ...      0      0   
3                              0         0        1  ...      0      0   
4                              0         0        0  ...      0      0   

   South American  Chicken  All Other  Mexican  price_level_1  price_level_2  \
0               0        0          0        0            0.0            1.0   
1               0        0          0        0            0.0            1.0   
2               0        0          0        0            0.0            1.0   
3               0        0          0        0            0.0            1.0   
4               0        0          0        0            0.0            1.0   

   price_level_3  price_level_4  
0            0.0            0.0  
1            0.0            0.0  
2            0.0            0.0  
3            0.0            0.0  
4            0.0            0.0  

[5 rows x 25 columns]

In [248]:
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import cross_validate

# define X and y
X = df_regression.drop(columns=['combined_rating','price_level'])
y = df_regression['combined_rating']

# Instantiate model 
model = LinearRegression() 

# 10-Fold Cross validate model 
cv_results = cross_validate(model, X, y, cv=10, scoring=['neg_mean_squared_error'])
cv_results

{'fit_time': array([0.01574492, 0.03063893, 0.01366401, 0.00649381, 0.02079177,
        0.00656796, 0.01476002, 0.00404692, 0.14500904, 0.00571012]),
 'score_time': array([0.00604081, 0.00257397, 0.00445914, 0.00400829, 0.00305605,
        0.00332022, 0.00517988, 0.00230694, 0.00338197, 0.02515197]),
 'test_neg_mean_squared_error': array([-0.0979626 , -0.08275897, -0.08192212, -0.1177427 , -0.09015041,
        -0.07207331, -0.06624049, -0.06311433, -0.07563396, -0.07640964])}

In [252]:
np.sqrt(abs(cv_results['test_neg_mean_squared_error'].mean()))

0.2870554861866533

In [234]:
cv_results

{'fit_time': array([0.24365211, 0.35412312, 0.19588518, 0.21962094, 0.24735999,
        0.22176313, 0.21953392, 0.19442606, 0.17393398, 0.18688011]),
 'score_time': array([0.01135802, 0.01688695, 0.00876808, 0.00840998, 0.02368784,
        0.00980186, 0.01097608, 0.00786114, 0.0079031 , 0.00931191]),
 'test_score': array([-0.19964774, -0.30692559, -0.18929564, -0.28026793,  0.03802218,
         0.0745098 , -0.01474201, -0.30585264, -0.239     , -0.48731219])}

In [227]:
from sklearn.neighbors import KNeighborsRegressor
from sklearn.ensemble import VotingRegressor, StackingRegressor, RandomForestRegressor, AdaBoostRegressor, GradientBoostingRegressor
from sklearn.model_selection import cross_val_score
from sklearn.tree import DecisionTreeRegressor
from xgboost import XGBRegressor

In [219]:
model = KNeighborsRegressor()
cv_results = cross_validate(model, X, y, cv=10)
score =  cv_results['test_score'].mean()

-0.12752565259075604

In [221]:
model = RandomForestRegressor(max_depth=50,min_samples_leaf=20)
cv_results = cross_validate(model, X, y, cv=10)
score =  cv_results['test_score'].mean()
score

-0.04460144320494508

In [224]:
model = XGBRegressor()
cv_results = cross_validate(model, X, y, cv=10)
score = cv_results['test_score'].mean()
score

-0.3514250071563606

In [226]:
model = AdaBoostRegressor(
    base_estimator=DecisionTreeRegressor(max_depth=None))

cv_results = cross_validate(model, X, y, cv=10)
score = cv_results['test_score'].mean()
score

-0.1910511756720533

In [245]:
X

geometry.location.lat  geometry.location.lng  Seafood  \
0                 0.484261               0.019853        0   
1                 0.307162               0.022085        0   
2                 0.594350               0.012711        0   
3                 0.598420               0.001937        0   
4                 0.807508               0.004836        0   
..                     ...                    ...      ...   
587               0.537853               0.912303        0   
588               0.219484               0.922133        0   
589               0.370427               0.991839        0   
590               0.367882               0.999133        0   
591               0.408762               1.000000        0   

     Vegan / Vegetarian / Salad  Middle Eastern  \
0                             0               0   
1                             0               0   
2                             0               0   
3                             0               0   
4                             0               0   
..                          ...             ...   
587                           0               0   
588                           0               0   
589                           0               0   
590                           0               0   
591                           0               0   

     Steakhouse / Barbecue / Grill  Hawaiian  General  American  Indian  ...  \
0                                0         0        0         0       0  ...   
1                                0         0        1         0       0  ...   
2                                0         0        1         0       0  ...   
3                                0         0        1         0       0  ...   
4                                0         0        0         0       0  ...   
..                             ...       ...      ...       ...     ...  ...   
587                              0         0        1         0       0  ...   
588                              0         0        1         0       0  ...   
589                              0         0        0         1       0  ...   
590                              0         0        0         0       0  ...   
591                              0         0        1         0       0  ...   

     Asian  Pizza  South American  Chicken  All Other  Mexican  price_level_1  \
0        0      0               0        0          0        0            0.0   
1        0      0               0        0          0        0            0.0   
2        0      0               0        0          0        0            0.0   
3        0      0               0        0          0        0            0.0   
4        0      0               0        0          0        0            0.0   
..     ...    ...             ...      ...        ...      ...            ...   
587      0      0               0        0          0        0            0.0   
588      0      0               0        0          0        0            0.0   
589      0      0               0        0          0        0            0.0   
590      0      0               0        0          0        0            0.0   
591      0      0               0        0          0        0            1.0   

     price_level_2  price_level_3  price_level_4  
0              1.0            0.0            0.0  
1              1.0            0.0            0.0  
2              1.0            0.0            0.0  
3              1.0            0.0            0.0  
4              1.0            0.0            0.0  
..             ...            ...            ...  
587            0.0            1.0            0.0  
588            1.0            0.0            0.0  
589            1.0            0.0            0.0  
590            1.0            0.0            0.0  
591            0.0            0.0            0.0  

[592 rows x 23 columns]

In [253]:
X.columns

Index(['geometry.location.lat', 'geometry.location.lng', 'Seafood',
       'Vegan / Vegetarian / Salad', 'Middle Eastern',
       'Steakhouse / Barbecue / Grill', 'Hawaiian', 'General', 'American',
       'Indian', 'European', 'Hamburger', 'African', 'Asian', 'Pizza',
       'South American', 'Chicken', 'All Other', 'Mexican', 'price_level_1',
       'price_level_2', 'price_level_3', 'price_level_4'],
      dtype='object')

In [255]:
# define X and y
X = df_regression.drop(columns=['combined_rating','price_level'])
y = df_regression['combined_rating']

# Instantiate model 
model = LinearRegression() 

reg = model.fit(X, y)

In [258]:
reg.score(X,y)

0.05935111087421663